pkg/build/starnix.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package build
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/syzkaller/pkg/osutil"
+	"github.com/google/syzkaller/sys/targets"
+	starnixVM "github.com/google/syzkaller/vm/starnix"
+)
+
+type starnix struct{}
+
+// The name of the Fuchsia assembly override defined in `overrideRule`.
+const overrideName = "syzkaller_starnix"
+
+// A Fuchsia assembly override definition adding a package containing fuzzing dependencies.
+var overrideRule = fmt.Sprintf(`import("//build/assembly/developer_overrides.gni")
```

Comment:
Would it possibly make sense to commit this file somewhere into the Fuchsia source instead? (Similar to the syzkaller-specific starnix component manifests).

Reply:
The current recommendation from the Fuchsia software assembly folks is to avoid checking in assembly overrides. We might be able to check in a custom product bundle definition -- there are some roughly similar definitions for other kinds of testing. Let's go with this for now, but I'll keep a TODO to figure out whether that's an option.

Reply:
Sounds good, thanks for looking into it!